src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Buttons/BitToggleButtonTests.cs

```diff
@@ -0,0 +1,195 @@
+
+            var label = (isChecked) ? checkedLabel : unCheckedLabel;
+            Assert.AreEqual(bitLabelTag.TextContent, label);
+
+            var iconName = (isChecked) ? checkedIconName : unCheckedIconName;
+            Assert.IsTrue(bitIconTag.ClassList.Contains($"bit-icon--{iconName}"));
+
+            return Task.CompletedTask;
+        }
+
+        [DataTestMethod,
+            DataRow(Visual.Fluent, true, true, "Button checked", "Button unchecked", "EmojiNeutral", "Emoji2", false),
+            DataRow(Visual.Fluent, true, false, "Button checked", "Button unchecked", "EmojiNeutral", "Emoji2", true),
+            DataRow(Visual.Fluent, false, true, "Button checked", "Button unchecked", "EmojiNeutral", "Emoji2", true),
+            DataRow(Visual.Fluent, false, false, "Button checked", "Button unchecked", "EmojiNeutral", "Emoji2", false),
+
+            DataRow(Visual.Cupertino, true, true, "Button checked", "Button unchecked", "EmojiNeutral", "Emoji2", false),
+            DataRow(Visual.Cupertino, true, false, "Button checked", "Button unchecked", "EmojiNeutral", "Emoji2", true),
+            DataRow(Visual.Cupertino, false, true, "Button checked", "Button unchecked", "EmojiNeutral", "Emoji2", true),
+            DataRow(Visual.Cupertino, false, false, "Button checked", "Button unchecked", "EmojiNeutral", "Emoji2", false),
+
+            DataRow(Visual.Material, true, true, "Button checked", "Button unchecked", "EmojiNeutral", "Emoji2", false),
+            DataRow(Visual.Material, true, false, "Button checked", "Button unchecked", "EmojiNeutral", "Emoji2", true),
+            DataRow(Visual.Material, false, true, "Button checked", "Button unchecked", "EmojiNeutral", "Emoji2", true),
+            DataRow(Visual.Material, false, false, "Button checked", "Button unchecked", "EmojiNeutral", "Emoji2", false)
+        ]
+        public Task BitToggleButtonClickEvent(Visual visual,
+            bool isChecked, bool isEnabled,
+            string checkedLabel, string unCheckedLabel, string checkedIconName, string unCheckedIconName,
+            bool expectedResult)
+        {
+            var component = RenderComponent<BitToggleButtonTest>(parameters =>
+            {
+                parameters.Add(p => p.Visual, visual);
+                parameters.Add(p => p.IsChecked, isChecked);
+                parameters.Add(p => p.ChekedLabel, checkedLabel);
+                parameters.Add(p => p.UnChekedLabel, unCheckedLabel);
+                parameters.Add(p => p.CheckedIconName, checkedIconName);
+                parameters.Add(p => p.UnCheckedIconName, unCheckedIconName);
+                parameters.Add(p => p.IsEnabled, isEnabled);
+            });
+
+            var bitToggleButton = component.Find(".bit-tgl-btn");
+
+            bitToggleButton.Click();
+
+            Assert.AreEqual(isEnabled ? 1 : 0, component.Instance.CurrentCount);
+
+            Assert.AreEqual(expectedResult, bitToggleButton.ClassList.Contains("bit-tgl-btn-checked"));
```

Comment:
instead of introducing the "expectedResult" argument, try to calculate this result by implementing the actual logic in this test method. for example, use the values of "isChecked" and "isEnabled" parameters.